include/remill/BC/Compat/PointerType.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2022 Trail of Bits, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "remill/BC/Version.h"
+
+namespace remill {
+
+inline static llvm::Type *getPointerElementType(llvm::PointerType *type) {
```

Comment:
Rename to `PointerElementType`. Is a checked needed for `type->isOpaque()` and have that return `nullptr`?

Reply:
Isn't it better to have the same name as LLVM does? So you don't have to always think whether it is with or without `get` prefix?

It is a precondition of both versions that the type is not opaque pointer. There is an assertion on that in both implementations.

Reply:
If you're putting it in the `llvm` namespace, use `get`. In the remill namespace, drop `get` or change to `Get`. Is it an assertion only enabled in debug builds, or all builds? If the former, make it return `nullptr`.

Reply:
Fixed in c0b87c1ec6b4d2ff8d627a703ec663b4cf0f7ba8 and 9e87c962b3fccec6a4897341aa7c747e99ecf041